Automattic/wp-calypso
# Jetpack Cloud: Add ability to restore

#### Changes proposed in this Pull Request

* Fill in restore flow with unstyled flow
* Hook up "Restore to this point" button to above flow
*  Several tweaks to jsdoc of selectors we use ( and typescript was complaining about )

##### 1. `/backups/:siteSlug` Start Restore
<img width="600" alt="start restore screenshot" src="https://user-images.githubusercontent.com/2810519/76011874-24154580-5eca-11ea-82e7-af69580a47ce.png">

##### 2. `/backups/:siteSlug/restore/:restoreId` Confirm Restore
<img width="600" alt="confirm restore screenshot" src="https://user-images.githubusercontent.com/2810519/76011924-3c856000-5eca-11ea-81ed-22a7fe400d9d.png">

##### 3. Restore In Progress

<img width="600" alt="restore in progress screenshot" src="https://user-images.githubusercontent.com/2810519/76012200-bf0e1f80-5eca-11ea-872c-3054365c3cbd.png">


##### 4. Restore Finished
<img width="600" alt="restore finished screenshot" src="https://user-images.githubusercontent.com/2810519/76011512-93d70080-5ec9-11ea-88b2-db87d2eda536.png">



#### Testing instructions



1. Navigate to `/backups/:siteSlug/restore/:restoreId` for a site with backups enables
2. Find a valid backup ( scrolling back in time if necessary ) 
3. Note the date you are backing up to and click "Restore to this point'
4. Verify you are taken to `/backups/:siteSlug/restore/:restoreId` and it matches the screenshot in step 2 above.
5. Click "Confirm Restore"
6. Verify that the page now displays "BackupRestoreQueued"
7. Verify that the page shortly switches to a progress bar matching step 3 above
8. Verify eventually the restore finished and matches the screenshot fo step 4 above
9. Navigate to `https://wordpress.com/activity-log/:siteSlug` for the same site
10. Verify there is a restore to the same date you noted in step 3
